### DISCUSSION TITLE
What's the difference between move.direction and move.angle?

### DISCUSSION THREAD

**Discussion by htopia** [matteobruni/tsparticles#2416] [Q&A] [answered]
Hi there,
I've been trying to make a rain effect and I keep getting the particles to go in different directions instead of the same direction.
I tried setting the angle in both the direction and angle properties of the move property but it keeps changing it randomly somehow.

Here's the pen where you can see the issue: https://codepen.io/htopia/pen/dyzRWxK

Any idea why this happens?

Thanks in advance.

**Comment by matteobruni** [comment, marked as answer]
Hi @htopia,

`angle` sets the spread angle of the movement for the randomization. If you have particles going down like in the sample, a smaller angle spread them less then a wider one.

`direction` is the direction of the particles, which can be a defined angle (numeric) or a general direction (`top`, `left`, etc.).

The particles movement has two other properties: `random` and `straight`. The first one multiplies the movement speed with a random factor, the second one interact with `angle`, adding to the direction angle a random value between `angle` min and `angle` max.

I'll explain `straight` better, it's a bit complicated:

- Set to `true`, the particle will follow the exact direction set in the config, for example if it's set to `left`, the particle will go horizontally left. If set to any number the particle will follow the exact direction set.
- Set to `false` (default), the particle will have a general direction read from the config, with a 90° spread (default). If you reduce `angle` values, the particles will have a smaller angle for spread. If you set the value to `0` it's the same as `straight: true`. The spread is calculated starting from the direction angle, which is the center of the spread angle, and a random value between the spread angle will be used.

**Reply by htopia** [reply]
hi @matteobruni, thanks for the great explanation!

So according to what you say, if I set `straight: true` if I want my rain particles to fall in a 45° angle direction, it should be just a matter of setting direction to 45 right?

I've updated my pen with these values, I also set `random: false` just in case and angle to 0 since I just want them all to fall exactly with a 45° angle, but they keep falling down straight without any angle.

Am I missing something? Is the numeric value in the direction property working as expected?

**Reply by matteobruni** [reply]
Yes, if you set `straight: true` they should follow your direction. You can also set `"bottom-left"` or `"bottom-right"` for 45° down-left or down-right.

`direction` numeric values replaces the `"bottom"` value, so you have to calculate the angle from an imaginary horizontal middle-right line which is 0°.

**Reply by htopia** [reply]
Ooh, I just noticed that I had another direction property on the emitters object which was set to bottom, I set the angle there and now it worked.
I thought that setting the direction on the move object would override the one in the emitters object.
